src/heroes/doomfist.opy

```diff
@@ -0,0 +1,138 @@
+#!include "ow2_constants.opy"
+
+
+rule "Doom shields":
+    @Event eachPlayer
+    @Hero doomfist
+    @Condition eventPlayer.getHealthOfType(Health.SHIELDS) == 100
+    
+    eventPlayer.setStatusEffect(eventPlayer, Status.UNKILLABLE, 9999)
+
+
+rule "Doom shields":
+    @Event eachPlayer
+    @Hero doomfist
+    @Condition eventPlayer.getHealthOfType(Health.SHIELDS) < 100
+    
+    eventPlayer.clearStatusEffect(Status.UNKILLABLE)
+
+
+rule "Speed":
+    @Event playerDealtDamage
+    @Hero doomfist
+    @Condition eventPlayer.isUsingAbility1() == true
+    
+    victim.setMoveSpeed(95)
+    wait(2.5)
+    victim.setMoveSpeed(100)
+
+
+rule "Knockback":
+    @Event eachPlayer
+    @Hero doomfist
+    @Condition isGameInProgress() == true
+    
+    eventPlayer.setKnockbackReceived(130)
+
+
+rule "More Knockback":
+    @Event eachPlayer
+    @Hero doomfist
+    @Condition eventPlayer.isUsingAbility1() == true
+    
+    eventPlayer.applyImpulse(Vector.DOWN, 10, Relativity.TO_WORLD, Impulse.INCORPORATE_CONTRARY_MOTION)
+    wait(0.45)
+
+
+rule "Upper cut":
+    @Event eachPlayer
+    @Hero doomfist
+    @Condition eventPlayer.isUsingAbility2() == true
+    
+    eventPlayer.applyImpulse(Vector.UP, 15, Relativity.TO_WORLD, Impulse.CANCEL_CONTRARY_MOTION)
+    getPlayersInRadius(eventPlayer, 5, getOppositeTeam(eventPlayer.getTeam()), LosCheck.OFF).applyImpulse(Vector.UP, 15, Relativity.TO_WORLD, Impulse.CANCEL_CONTRARY_MOTION)
+    #getPlayersInRadius(eventPlayer, 5, getOppositeTeam(eventPlayer.getTeam()), LosCheck.OFF).setStatusEffect(eventPlayer, Status.STUNNED, 1.25)
+    damage(getPlayersInRadius(eventPlayer, 5, getOppositeTeam(eventPlayer.getTeam()), LosCheck.OFF), eventPlayer, 50)
+    getPlayersInRadius(eventPlayer, 5, getOppositeTeam(eventPlayer.getTeam()), LosCheck.OFF).setMoveSpeed(0)
+    eventPlayer.cancelPrimaryAction()
+    wait(0.95)
+    eventPlayer.setGravity(0)
+    getPlayersInRadius(eventPlayer, 5, getOppositeTeam(eventPlayer.getTeam()), LosCheck.OFF).setStatusEffect(eventPlayer, Status.ROOTED, 0.6)
+    getPlayersInRadius(eventPlayer, 5, getOppositeTeam(eventPlayer.getTeam()), LosCheck.OFF).setGravity(0)
+    wait(0.6)
+    getAllPlayers().setGravity(100)
+    getAllPlayers().setMoveSpeed(100)
+
+
+rule "DOOMS FIST":
+    @Event eachPlayer
+    @Hero doomfist
+    @Condition eventPlayer.isFiringSecondaryFire() == true
+    
+    eventPlayer.setDamageDealt(370)
+    wait(0.15)
+    eventPlayer.setDamageDealt(370)
```

Comment:
Magic numbers; should be replaced with keywords from ow1/ow2_constants.opy. I'll fix these magic numbers myself if you don't want to do it.

Reply:
Ill fix this later